src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VUAP.java

```diff
@@ -19,19 +18,19 @@
   private WebElement iframe;
   private WebElement playTriggerButton;
 
-  public VUAP(WikiaWebDriver driver, String slotName) {
+  public VUAP(WikiaWebDriver driver, String platform, String slotName) {
     this.wait = new Wait(driver);
     this.driver = driver;
-    setIframe(slotName);
+    setIframe(platform, slotName);
     setTriggerButton(driver);
   }
 
-  private String getAdUnit(String slotName) {
-    return "google_ads_iframe_/5441/wka.life/_project43//article/gpt/" + slotName + "_0";
+  private String getAdUnit(String platform, String slotName) {
+    return "google_ads_iframe_/5441/wka.life/_project43//article/" + platform + "/" + slotName + "_0";
```

Comment:
Have you checked if we don't have similar code in our classes which you could re-use?